pubsub/alicloud/confluentkafka/confluentkafka.go

```diff
@@ -0,0 +1,266 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package confluentkafka
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/kit/logger"
+)
+
+var (
+	ProducerFlushTimeoutMs = 10 * 1000
+)
+
+type contextWithCancelFunc struct {
+	Ctx        context.Context
+	CancelFunc context.CancelFunc
+}
+
+type Kafka struct {
+	ConfigMap *kafka.ConfigMap
+	logger    logger.Logger
+	producer  *kafka.Producer
+	consumer  *kafka.Consumer
+	topics    []string
+
+	consumerContext contextWithCancelFunc
+}
+
+func NewConfluentKafka(logger logger.Logger) pubsub.PubSub {
+	return &Kafka{ //nolint:exhaustivestruct
+		logger: logger,
+	}
+}
+
+func (k *Kafka) parseMetadata(props map[string]string) error {
+	k.ConfigMap = &kafka.ConfigMap{}
+
+	var metadata map[string]interface{}
+	if configJson, ok := props["configJson"]; ok {
+		err := json.Unmarshal([]byte(configJson), &metadata)
+		if err != nil {
+			return err
+		}
+
+		for key, val := range metadata{
+			if reflect.ValueOf(val).Kind() == reflect.Float64{
```

Comment:
Instead of using the `reflect` package, this would be simpler:

```golang
if v, ok := val.(float64); ok && v == math.Trunc(v) {
  val = int(v)
}
k.ConfigMap.SetKey(key, v))
```

Reply:
modified